controllers/keywordsController.js

```diff
@@ -0,0 +1,25 @@
+const { isValidString, isEmptyString } = require("../utils/validation");
+const keywordModel = require("../models/keywordModel");
+const postModel = require("../models/postModel");
+
+const list = async (req, res) => {
+  if (!isValidString(req.params.keywordId) || isEmptyString(req.params.keywordId)) {
+    return res.status(400).send({ message: "[InvalidKeyword] Error occured" });
+  }
+
+  try {
+    const { keywordId } = req.params;
+    const keywordResult = await keywordModel.findById(keywordId).exec();
+
+    const postList = await postModel.find({ keywordId }).exec();
+    const postIdList = postList.length === 0 ? [] : postList.map((data) => data._id);
+
+    res.status(200).json({ ...keywordResult._doc, postId: postIdList });
```

Comment:
```suggestion
    res.status(200).json({ ...keywordResult, postId: postIdList });
```
`keywordResult._doc`에만 데이터 내용이 담겨져있나 라는 의문에 테스트를 진행해봤는데, `keywordResult`에도 동일하게 데이터 내용이 출력되었어요. 규경님이 개발 진행할 때는 데이터가 동일하게 출력되지 않아 `._doc` 를 붙이신 것으로 판단되는데, 제가 정적 데이터로 테스트를 해서 동일하게 출력되는 건지.. 해당 부분 같이 확인해보면 좋을 것 같아요! 
cc @jin-ttao 

우선 제가 테스트 해본 결과 스크린샷 첨부드립니다!
![스크린샷 2024-11-11 오전 11 44 11](https://github.com/user-attachments/assets/c0af1d9e-ffc6-4ca1-ab89-fe9e57ff578b)

Reply:
저도 말씀주신대로 keywordResult로 하고 싶었는데요, 한 번 객체에 `keywordResult`를 구조분해 할당 해보시면 보이지 않는 프로퍼티들이 보이는 것을 확인하실 수 있으실거에요.

`console.log({ ...keywordResult });`의 결과 입니다 !

![image](https://github.com/user-attachments/assets/4e86feab-e12c-499c-8050-78dff82a6af6)

Reply:
구조분해할당을 하면서 구조가 달라지는거군요..! 
해당 내용에 대해서 검색을 해보니, 구조분해할당을 할 때 `._doc` 를 사용할 수도 있고 또는 `.toObject()`를 사용할 수도 있다고 하네요! [링크](https://saiphani.hashnode.dev/how-to-deletehide-properties-of-mongoose-object-nodejs)

`.toObject()`를 테스트해본 결과도 첨부합니다.
![스크린샷 2024-11-11 오후 2 05 10](https://github.com/user-attachments/assets/47d9a017-0a50-4edb-b26c-9ca9a33a88b3)

Reply:
오 ! 정보 감사합니다 ! mongoose에서 공식적으로 지원하는 `toObject()` 메소드를 쓰는게 더 좋아보이네요 :)